include/fiction/utils/layout_utils.hpp

```diff
@@ -350,6 +353,36 @@ Lyt convert_to_fiction_coordinates(const sidb_cell_clk_lyt_siqad& lyt) noexcept
 
     return lyt_new;
 }
+/**
+ * Generates a random coordinate within the region spanned by two given coordinates. The two given coordinates form the
+ * top left corner and the bottom right corner of the spanned region.
+ *
+ * @OffsetCoordinateType The coordinate implementation to be used.
+ * @param coordinate1 Top left Coordinate.
+ * @param coordinate2 Bottom right Coordinate (coordinate order is not important, automatically swapped if
+ * necessary).
+ * @return Randomly generated coordinate.
+ */
+template <typename CoordinateType>
+CoordinateType random_coordinate(CoordinateType coordinate1, CoordinateType coordinate2) noexcept
+{
+    static_assert(std::is_same_v<CoordinateType, offset::ucoord_t> || std::is_same_v<CoordinateType, cube::coord_t> ||
+                      std::is_same_v<CoordinateType, siqad::coord_t>,
+                  "CoordinateType is unknown");
+
+    static std::mt19937_64 generator(std::random_device{}());
+
+    if (coordinate1 > coordinate2)
+    {
+        std::swap(coordinate1, coordinate2);
+    }
+
+    std::uniform_int_distribution<> dist_x(coordinate1.x, coordinate2.x);
+    std::uniform_int_distribution<> dist_y(coordinate1.y, coordinate2.y);
+    std::uniform_int_distribution<> dist_z(coordinate1.z, coordinate2.z);
```

Comment:
I think this might break for SiQAD coordinates. If `coordinate2` is not located on a `z = 1` dimer atom, this function will always produce lattice coordinates on the `z = 0` dimer atoms. Is this intended behavior?